examples/08_join_aggregation.py

```diff
@@ -1,302 +1,311 @@
+# "make" and "goods_code" on the other hand consist in orthogonal categories. We
+# encode them using a OrdinalEncoder.
 #
-# This feature answer questions like
-# *"How many times has this user given a bad, medium or good rate to movies?"*.
-from skrub import AggTarget
-
-agg_target_user = AggTarget(
-    main_key="userId",
-    suffix="_user",
-    operation="hist(3)",
-)
-X_transformed = agg_target_user.fit_transform(X, y)
-
-X_transformed.shape
-###############################################################################
-X_transformed.head()
-
-###############################################################################
-# Similarly, we join on ``movieId`` instead of ``userId``.
+# Finally, we bring this logic into a TableVectorizer to encode these columns in a
+# single step.
 #
-# This feature answer questions like
-# *"How many times has this movie received a bad, medium or good rate from users?"*.
-agg_target_movie = AggTarget(
-    main_key="movieId",
-    suffix="_movie",
-    operation="hist(3)",
-)
-X_transformed = agg_target_movie.fit_transform(X, y)
-X_transformed.shape
-###############################################################################
-X_transformed.head()
-
-###############################################################################
-# Chaining everything together in a pipeline
-# ------------------------------------------
-#
-# To perform cross-validation and enable hyper-parameter tuning, we gather
-# all elements into a scikit-learn |Pipeline| by using |make_pipeline|,
-# and define a scikit-learn |HGBR|.
-from sklearn.ensemble import HistGradientBoostingRegressor
-from sklearn.pipeline import make_pipeline
+# See `here <https://skrub-data.org/stable/auto_examples/01_encodings.html#specializing-the-tablevectorizer-for-histgradientboosting>`_
+# for more details about these choices.
+from sklearn.preprocessing import OrdinalEncoder
 
-pipeline = make_pipeline(
-    table_vectorizer,
-    agg_target_user,
-    agg_target_movie,
-    HistGradientBoostingRegressor(learning_rate=0.1, max_depth=4, max_iter=40),
-)
+from skrub import MinHashEncoder, TableVectorizer
 
-pipeline
+vectorizer = TableVectorizer(
+    high_cardinality=MinHashEncoder(),  # encode ["item", "model"]
+    specific_transformers=[
+        (OrdinalEncoder(), ["make", "goods_code"]),
+    ],
+)
+products_transformed = vectorizer.fit_transform(products)
+TableReport(products_transformed)
 
-###############################################################################
-# Hyper-parameters tuning and cross validation
-# --------------------------------------------
+# %%
+# Our objective is now to aggregate this dataframe by basket ID, then merging it
+# on the basket dataframe, still on the basket ID.
+#
+# AggJoiner can help us achieve exactly this. We need to pass the product dataframe as
+# an auxiliary table argument to AggJoiner in ``__init__``. ``aux_key`` represent both
```

Comment:
I'm not sure I'd mention ``__init__``, do we expect people to know this ? Completely fine with leaving it if we do